src/nix/develop.cc

```diff
@@ -507,7 +507,8 @@ struct CmdDevelop : Common, MixEnvironment
             std::vector<std::string> args;
             for (auto s : command)
                 args.push_back(shellEscape(s));
-            script += fmt("exec %s\n", concatStringsSep(" ", args));
+            auto argString = concatStringsSep(" ", args);
+            script += fmt("if [ \"$(type -t %s)\" = file ]; then exec %s; else %s; fi\n", args[0], argString, argString);
```

Comment:
I don't know whether we want to go with this change at all, but if we do, I'd rather not have this bit of magic and just have.

```suggestion
            script += argString;
```

The `exec` call isn't that useful in 95% of the cases and can always be added manually if it's really needed